internal/providers/vsphere/pke/workflow/create_node_activity.go

```diff
@@ -0,0 +1,276 @@
+// Copyright © 2019 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package workflow
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"encoding/base64"
+	"fmt"
+	"strings"
+	"text/template"
+
+	"emperror.dev/errors"
+	"github.com/ghodss/yaml"
+	"github.com/vmware/govmomi/find"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25/types"
+	"go.uber.org/cadence/activity"
+
+	"github.com/banzaicloud/pipeline/internal/providers/pke/pkeworkflow/pkeworkflowadapter"
+)
+
+// CreateNodeActivityName is the default registration name of the activity
+const CreateNodeActivityName = "pke-vsphere-create-node"
+
+// CreateNodeActivity represents an activity for creating a vSphere virtual machine
+type CreateNodeActivity struct {
+	vmomiClientFactory *VMOMIClientFactory
+	tokenGenerator     pkeworkflowadapter.TokenGenerator
+}
+
+// MakeCreateNodeActivity returns a new CreateNodeActivity
+func MakeCreateNodeActivity(vmomiClientFactory *VMOMIClientFactory, tokenGenerator pkeworkflowadapter.TokenGenerator) CreateNodeActivity {
+	return CreateNodeActivity{
+		vmomiClientFactory: vmomiClientFactory,
+		tokenGenerator:     tokenGenerator,
+	}
+}
+
+// CreateNodeActivityInput represents the input needed for executing a CreateNodeActivity
+type CreateNodeActivityInput struct {
+	OrganizationID   uint
+	ClusterID        uint
+	SecretID         string
+	ClusterName      string
+	ResourcePoolName string
+	FolderName       string
+	DatastoreName    string
+	Node
+}
+
+// Node represents a vSphere virtual machine
+type Node struct {
+	AdminUsername          string
+	VCPU                   int
+	RamMB                  int
```

Comment:
`Ram`, but it should be `RAM`: https://go.googlesource.com/lint/+/d0100b6bd8b389f0385611eb39152c4d7c3a7905/lint.go#788